src/main/kotlin/no/nav/familie/ef/sak/service/VedtaksbrevService.kt

```diff
@@ -0,0 +1,43 @@
+package no.nav.familie.ef.sak.service
+
+import no.nav.familie.ef.sak.api.dto.BrevRequest
+import no.nav.familie.ef.sak.brev.BrevClient
+import no.nav.familie.ef.sak.repository.VedtaksbrevRepository
+import no.nav.familie.ef.sak.repository.domain.Vedtaksbrev
+import no.nav.familie.ef.sak.service.steg.StegType
+import org.springframework.stereotype.Service
+import java.util.*
+
+@Service
+class VedtaksbrevService(private val brevClient: BrevClient,
+                         private val brevRepository: VedtaksbrevRepository,
+                         private val behandlingService: BehandlingService,
+                         private val fagsakService: FagsakService,
+                         private val personService: PersonService) {
+
+    fun lagBrev(behandlingId: UUID): ByteArray {
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        val fagsak = fagsakService.hentFagsak(behandling.fagsakId)
+        val person = personService.hentSøker(fagsak.hentAktivIdent())
+        val navn = person.navn.get(0)
+
+        val request = BrevRequest(navn = "${navn.fornavn} ${navn.etternavn}", ident = fagsak.hentAktivIdent())
+
+        val brev = brevRepository.insert(Vedtaksbrev(behandlingId = behandlingId,
+                                                     brevRequest = request,
+                                                     steg = behandling.steg,
+                                                     pdf = brevClient.genererBrev("bokmaal",
+                                                                                  "testDokument",
+                                                                                  request)))
+        return brev.pdf
+    }
+
+    fun hentBrev(behandlingId: UUID): Vedtaksbrev? {
+        return brevRepository.findByBehandlingId(behandlingId).first()
+    }
+
+    fun hentBrevForTotrinnskontroll(behandlingId: UUID): Vedtaksbrev? {
+        return brevRepository.findByBehandlingId(behandlingId).firstOrNull { it.steg === StegType.BESLUTTE_VEDTAK }
```

Comment:
Når er det tenkt at disse skal brukes forskjellig?

Reply:
Det er to forskjellige caser for henting av brev:
1. Saksbehandler ønsker å hente opp et brev de tidligere har jobbet på for en behandling, men som enda ikke er sendt til beslutter.
2. En annen saksbehandler skal hente brevet for totrinnskontroll

I utgangspunktet tenkte jeg at det burde være to forskjellige funksjoner, men tror kanskje vi dropper `hentBrevForTotrinnskontroll`. Hvilket brev man skal ha vil jo være implisitt utifra behandlingens stegtype, og derfor kan vi ta den sjekken i `hentBrev`